content/2019/vim-and-the-shell/index.md

```diff
@@ -0,0 +1,227 @@
+---
+title: "Vim Filters, External Commands, And Fraternity"
+publishDate: 2019-12-01
+draft: true
+description: "Vim is not an isolated tool. It is designed to work with external programs."
+author:
+  email: "pabloariasal@gmail.com"
+  github: "pabloariasal "
+  homepage: "https://pabloariasal.github.io"
+  irc: "username"
+  name: "Pablo Arias"
+  picture: "https://example.com/username.jpg"
+  twitter: "@pabloariasal"
+---
+
+> Don't love me, kid. Love your mom. I just play football.
+>
+> – Juan Roman Riquelme
+
+There is a day in the life of a man when he realizes that the traditional way of using a computer is flawed, that graphical interfaces were a bad idea, that the mouse is an absurd invention, that a life away from the home row is not a life worth living.
+
+The UNIX philosophy becomes a religion when you are hit with the realization that best solutions are not provided by gigantic almighty monoliths, but are the joint effort of several minimal, highly specialized components. Each solves one part of the problem, just one, but solves it well.
+
+In this post I'll explore the idea that vim is just another member of the GNU/Linux constellation of tools, perfectly integrated in its environment and the underlying shell. Vim's built-in interaction capabilities with external programs allow for very interesting synergies. Interplays that become particularly powerful in conjunction with advanced text processing utilities and filters
```

Comment:
Vim also works on BSDs, BSD derivatives like MacOS, Windows, and, let's not forget it, was created for the Amiga. I think the "GNU/Linux" part should be rephrased in a more universal way.

Reply:
yes, very good point. I've redesign the article to read more system-agnostic.